temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowWorker.java

```diff
@@ -62,22 +62,22 @@ public SyncWorkflowWorker(
       String namespace,
       String taskQueue,
       WorkerInterceptor[] workerInterceptors,
-      SingleWorkerOptions workflowOptions,
+      SingleWorkerOptions singleWorkerOptions,
       SingleWorkerOptions localActivityOptions,
       WorkflowExecutorCache cache,
       String stickyTaskQueueName,
       Duration stickyWorkflowTaskScheduleToStartTimeout,
       ThreadPoolExecutor workflowThreadPool) {
     Objects.requireNonNull(workflowThreadPool);
-    this.dataConverter = workflowOptions.getDataConverter();
+    this.dataConverter = singleWorkerOptions.getDataConverter();
 
     factory =
         new POJOWorkflowImplementationFactory(
-            workflowOptions.getDataConverter(),
+            singleWorkerOptions.getDataConverter(),
             workflowThreadPool,
             workerInterceptors,
             cache,
-            workflowOptions.getContextPropagators());
+            singleWorkerOptions);
```

Comment:
If the data converter is always taken from the worker options when this is called, we could eliminate it as a separate parameter.